src/main/extension/extension_load.cpp

```diff
@@ -132,6 +132,8 @@ void ExtensionHelper::LoadExternalExtension(ClientContext &context, const string
 		throw InvalidInputException("Initialization function \"%s\" from file \"%s\" threw an exception: \"%s\"",
 		                            init_fun_name, res.filename, e.what());
 	}
+
+	DuckDB(DatabaseInstance::GetDatabase(context)).SetExtensionLoaded(extension);
```

Comment:
Can we perhaps move the `SetExtensionLoaded` function to the `DatabaseInstance` instead?